coalib/core/DependencyTracker.py

```diff
@@ -203,32 +203,19 @@ def resolve(self, dependency):
         # Check if dependency has itself dependencies which aren't resolved,
         # these need to be removed too. This operation does not free any
         # dependencies.
-        dependencies_to_remove = []
-        for tracked_dependency, dependants in self._dependency_dict.items():
+        for tracked_deps, dependants in tuple(self._dependency_dict.items()):
```

Comment:
why do you tupelize it before? this makes just worse performance^^
(also a rename is not necessary imo)